feature/bgp/addpath/ate_tests/addpath/README.md

```diff
@@ -0,0 +1,91 @@
+## Summary
+This test checks for DUT functionality of BGP addpath capability. It also confirms accuracy of addpath capabilities negotiation, following checks
+1. DUT sets value for the Send/Receive field as "3" during capabilities negotiation to signal support for both Send and Receive ability for multiple paths.
+2. DUT sets value for the Send/Receive field as "1" during capabilities negotiation to signal support for Receive ability for multiple paths.
+3. DUT sets value for the Send/Receive field as "2" during capabilities negotiation to signal support for Send ability for multiple paths.
+4. Tests are conducted at scale of 1M v4 and 600k v6 routes with 64 different NHs each. Also, routes are withdrawn at scale to ensure addpath functionality is maintained at scale. Multipath is also enabled on the DUT to extend addpath functionalities implications on FIB programming at scale. 
+
+## Topology:
+ATE1:Port1 <-EBGP-> DUT:Port1
+ATE2:Port2 <-EBGP-> DUT:Port2
+ATE3:Port3 <-IBGP-> DUT:Port3
+ATE4:Port4 <-IBGP-> DUT:Port4
+ATE5:Port5 <-IBGP-> DUT:Port5
+
+### Test procedure
+Test-1 and Test-2 below are expected to be run at the global, peer-group and neighbor levels.
+
+**Test-1**: Verify ADDPATH Send/Receive capability
+* DUT:Port1 and DUT:Port2 has EBGP peering with ATE1:Port1 and ATE2:Port2 respectively.
+* DUT:Port3 and DUT:Port4 has IBGP peering with directly connected ATE3:Port3 and ATE4:Port4 respectively. In this case, DUT is the RR server and ATE3 and ATE4 are RR clients
+* DUT:Port5 has IBGP peering with directly connected ATE5:Port5. In this case DUT is the RR client and ATE5:Port5 is the RR server
+* Ensure that the EBGP and IBGP peering are enabled for both address famailies.
+* DUT should be configured with "Both" Send and Receive ability for addpath on all IBGP peering. Same for ATE3:Port3, ATE4:Port4 and ATE5:Port5
+  * During BGP capabilities negotiation in Open message, verify that the DUT negotiated addpath cability with Send/Receive field set to "3"
+
+* Configure ATE1:Port1 and ATE2:Port2 to advertise same prefix "prefix-1" with different Protocol next-hops to DUT:Port1 and DUT:Port2
+  * Verify that the DUT is advertising multiple paths to prefix-1 to RRCs ATE3 and ATE4 as well as to the RRS ATE5 with different path-ids
+* Configure each of ATE1:Port1 and ATE2:Port2 to advertise prefix "prefix-1" with 2 different Protocol next-hops to DUT:Port1 and DUT:Port2 respectively. Then configure DUT to send maximum 3 paths over ADD-Path to ATE3 and ATE4
```

Comment:
```suggestion
* Configure each of ATE1:Port1 and ATE2:Port2 to advertise to path each for prefix "prefix-1" with 2 different Protocol next-hops to DUT:Port1 and DUT:Port2 respectively. Then configure DUT to send maximum 3 paths over ADD-Path to ATE3 and ATE4
```

Reply:
I made some modifications and hopefully this adds more clarity